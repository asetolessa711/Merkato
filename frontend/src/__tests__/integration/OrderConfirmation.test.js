import React from 'react';
import { render, screen } from '@testing-library/react';

describe('OrderConfirmation', () => {
  it('renders without crashing', () => {
    // Replace with actual OrderConfirmation component when available
    expect(true).toBe(true);
  });
});

describe('\ud83e\uddfe OrderConfirmation Page', () => {
  test('displays confirmation message and order details', () => {
    render(
      <MemoryRouter>
        <OrderConfirmation />
      </MemoryRouter>
    );

    expect(screen.getByText(/thank you/i)).toBeInTheDocument();
    expect(screen.getByText(/your order has been placed/i)).toBeInTheDocument();
    // If your page displays order number or summary, check for it:
    // expect(screen.getByText(/order #\d+/i)).toBeInTheDocument();
  });

  test('offers link or button to return to homepage or shop', () => {
    render(
      <MemoryRouter>
        <OrderConfirmation />
      </MemoryRouter>
    );
    // Accept either a link or button for navigation
    const shopLink = screen.queryByRole('link', { name: /shop|home/i });
    const shopButton = screen.queryByRole('button', { name: /shop|home/i });
    expect(shopLink || shopButton).toBeInTheDocument();
  });

  test('has a confirmation heading', () => {
    render(
      <MemoryRouter>
        <OrderConfirmation />
      </MemoryRouter>
    );
    expect(screen.getByRole('heading', { name: /order confirmation/i })).toBeInTheDocument();
  });

  test('navigates to shop/home when link or button is clicked', () => {
    render(
      <MemoryRouter>
        <OrderConfirmation />
      </MemoryRouter>
    );
    const shopLink = screen.queryByRole('link', { name: /shop|home/i });
    const shopButton = screen.queryByRole('button', { name: /shop|home/i });
    if (shopLink) {
      // ...simulate click and assert navigation...
    }
    if (shopButton) {
      // ...simulate click and assert navigation...
    }
  });
});
