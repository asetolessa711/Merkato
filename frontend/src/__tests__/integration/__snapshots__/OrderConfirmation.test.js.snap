// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`üßæ OrderConfirmation Page matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="max-width: 700px; margin: 40px auto; background: rgb(255, 255, 255); border-radius: 10px; box-shadow: 0 2px 10px rgba(0,0,0,0.08); padding: 32px; font-family: Poppins, sans-serif;"
  >
    <div
      style="text-align: center; margin-bottom: 16px;"
    >
      <h2
        aria-level="2"
        role="heading"
        style="color: rgb(0, 184, 148); margin-bottom: 8px;"
      >
        Order Confirmation
      </h2>
      <div
        style="font-size: 22px; font-weight: 600; color: rgb(0, 184, 148); margin-bottom: 4px;"
      >
        Thank you!
      </div>
      <div
        style="font-size: 16px; color: rgb(85, 85, 85);"
      >
        Your order has been placed successfully.
      </div>
    </div>
    <div
      style="margin-bottom: 16px; color: rgb(85, 85, 85);"
    >
      <strong>
        Order Number:
      </strong>
       12345
      <br />
      <strong>
        Invoice Number:
      </strong>
       INV-67890
      <br />
      <strong>
        Date:
      </strong>
       2025-08-05
    </div>
    <div
      style="display: flex; gap: 32px; margin-bottom: 24px;"
    >
      <div
        style="flex: 1 1 0%;"
      >
        <h4
          aria-level="4"
          role="heading"
          style="margin-bottom: 4px;"
        >
          Buyer Info (guest):
        </h4>
        <div>
          Name: Guest Buyer
        </div>
        <div>
          Email: guest@example.com
        </div>
        <div>
          Phone: 1234567890
        </div>
        <div>
          Address: 123 Cypress Lane
        </div>
      </div>
      <div
        style="flex: 1 1 0%;"
      >
        <h4
          aria-level="4"
          role="heading"
          style="margin-bottom: 4px;"
        >
          Company Info:
        </h4>
        <div>
          Name: Test Company
        </div>
        <div>
          Address: 456 Company St
        </div>
        <div>
          Email: company@example.com
        </div>
        <div>
          Phone: 
          <span
            style="color: rgb(170, 170, 170);"
          >
            Not provided
          </span>
        </div>
      </div>
    </div>
    <table
      aria-label="Order Items"
      style="width: 100%; border-collapse: collapse; margin-bottom: 24px;"
    >
      <thead>
        <tr
          style="background: rgb(244, 244, 244);"
        >
          <th
            style="text-align: left; padding: 8px;"
          >
            Item
          </th>
          <th
            style="text-align: center; padding: 8px;"
          >
            Qty
          </th>
          <th
            style="text-align: right; padding: 8px;"
          >
            Price
          </th>
          <th
            style="text-align: right; padding: 8px;"
          >
            Total
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            style="padding: 8px;"
          >
            Cypress Test Product
          </td>
          <td
            style="text-align: center; padding: 8px;"
          >
            1
          </td>
          <td
            style="text-align: right; padding: 8px;"
          >
            USD 24.99
          </td>
          <td
            style="text-align: right; padding: 8px;"
          >
            USD 24.99
          </td>
        </tr>
      </tbody>
    </table>
    <div
      style="text-align: right; margin-bottom: 8px;"
    >
      <div>
        Subtotal: USD 20.00
      </div>
      <div>
        Tax: USD 4.99
      </div>
      <div
        style="font-weight: bold; font-size: 18px;"
      >
        Total: USD 24.99
      </div>
    </div>
    <div
      style="text-align: center; margin-top: 32px;"
    >
      <button
        aria-label="Print Invoice"
        style="background: rgb(0, 184, 148); color: rgb(255, 255, 255); padding: 10px 24px; border-radius: 6px; font-size: 16px; cursor: pointer; margin-bottom: 16px;"
      >
        üñ®Ô∏è Print Invoice
      </button>
      <br />
      <a
        aria-label="Return to Home"
        href="/"
        role="button"
        style="display: inline-block; margin-top: 12px; background: rgb(9, 132, 227); color: rgb(255, 255, 255); padding: 10px 24px; border-radius: 6px; font-size: 16px; text-decoration: none; cursor: pointer;"
      >
        üè† Return to Home
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`üßæ OrderConfirmation Page matches snapshot with different data 1`] = `
<DocumentFragment>
  <div
    style="max-width: 700px; margin: 40px auto; background: rgb(255, 255, 255); border-radius: 10px; box-shadow: 0 2px 10px rgba(0,0,0,0.08); padding: 32px; font-family: Poppins, sans-serif;"
  >
    <div
      style="text-align: center; margin-bottom: 16px;"
    >
      <h2
        aria-level="2"
        role="heading"
        style="color: rgb(0, 184, 148); margin-bottom: 8px;"
      >
        Order Confirmation
      </h2>
      <div
        style="font-size: 22px; font-weight: 600; color: rgb(0, 184, 148); margin-bottom: 4px;"
      >
        Thank you!
      </div>
      <div
        style="font-size: 16px; color: rgb(85, 85, 85);"
      >
        Your order has been placed successfully.
      </div>
    </div>
    <div
      style="margin-bottom: 16px; color: rgb(85, 85, 85);"
    >
      <strong>
        Order Number:
      </strong>
       99999
      <br />
      <strong>
        Invoice Number:
      </strong>
       INV-67890
      <br />
      <strong>
        Date:
      </strong>
       2025-08-05
    </div>
    <div
      style="display: flex; gap: 32px; margin-bottom: 24px;"
    >
      <div
        style="flex: 1 1 0%;"
      >
        <h4
          aria-level="4"
          role="heading"
          style="margin-bottom: 4px;"
        >
          Buyer Info (guest):
        </h4>
        <div>
          Name: Guest Buyer
        </div>
        <div>
          Email: guest@example.com
        </div>
        <div>
          Phone: 1234567890
        </div>
        <div>
          Address: 123 Cypress Lane
        </div>
      </div>
      <div
        style="flex: 1 1 0%;"
      >
        <h4
          aria-level="4"
          role="heading"
          style="margin-bottom: 4px;"
        >
          Company Info:
        </h4>
        <div>
          Name: Test Company
        </div>
        <div>
          Address: 456 Company St
        </div>
        <div>
          Email: company@example.com
        </div>
        <div>
          Phone: 
          <span
            style="color: rgb(170, 170, 170);"
          >
            Not provided
          </span>
        </div>
      </div>
    </div>
    <table
      aria-label="Order Items"
      style="width: 100%; border-collapse: collapse; margin-bottom: 24px;"
    >
      <thead>
        <tr
          style="background: rgb(244, 244, 244);"
        >
          <th
            style="text-align: left; padding: 8px;"
          >
            Item
          </th>
          <th
            style="text-align: center; padding: 8px;"
          >
            Qty
          </th>
          <th
            style="text-align: right; padding: 8px;"
          >
            Price
          </th>
          <th
            style="text-align: right; padding: 8px;"
          >
            Total
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            style="padding: 8px;"
          >
            Special Product
          </td>
          <td
            style="text-align: center; padding: 8px;"
          >
            3
          </td>
          <td
            style="text-align: right; padding: 8px;"
          >
            USD 99.99
          </td>
          <td
            style="text-align: right; padding: 8px;"
          >
            USD 299.97
          </td>
        </tr>
      </tbody>
    </table>
    <div
      style="text-align: right; margin-bottom: 8px;"
    >
      <div>
        Subtotal: USD 299.97
      </div>
      <div>
        Tax: USD 0.00
      </div>
      <div
        style="font-weight: bold; font-size: 18px;"
      >
        Total: USD 299.97
      </div>
    </div>
    <div
      style="text-align: center; margin-top: 32px;"
    >
      <button
        aria-label="Print Invoice"
        style="background: rgb(0, 184, 148); color: rgb(255, 255, 255); padding: 10px 24px; border-radius: 6px; font-size: 16px; cursor: pointer; margin-bottom: 16px;"
      >
        üñ®Ô∏è Print Invoice
      </button>
      <br />
      <a
        aria-label="Return to Home"
        href="/"
        role="button"
        style="display: inline-block; margin-top: 12px; background: rgb(9, 132, 227); color: rgb(255, 255, 255); padding: 10px 24px; border-radius: 6px; font-size: 16px; text-decoration: none; cursor: pointer;"
      >
        üè† Return to Home
      </a>
    </div>
  </div>
</DocumentFragment>
`;
